gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/cluster/BrokerTopologyManagerImpl.java

```diff
@@ -213,22 +198,15 @@ private void updateMetrics(final BrokerClusterState topology) {
   }
 
   @Override
-  public void onClusterTopologyChanged(final ClusterTopology clusterTopology) {
-    actor.run(
-        () -> {
-          if (clusterTopology.isUninitialized()) {
-            return;
-          }
-
-          final BrokerClusterStateImpl newTopology = new BrokerClusterStateImpl(topology.get());
-
-          // Overwrite clusterSize. ClusterTopology is the source of truth.
-          newTopology.setClusterSize(clusterTopology.clusterSize());
-          newTopology.setPartitionsCount(clusterTopology.partitionCount());
-
-          LOG.debug(
-              "Received new cluster topology with clusterSize {}", newTopology.getClusterSize());
-          topology.set(newTopology);
+  public void onTopologyUpdated(final ClusterTopology clusterTopology) {
+    if (clusterTopology.isUninitialized()) {
+      return;
+    }
+    LOG.debug("Received new cluster topology with clusterSize {}", clusterTopology.clusterSize());
+    updateTopology(
+        topology -> {
```

Comment:
:upside_down_face: My IDE shows warning `Rename "topology" which hides the field declared at line 32.`